lib/dry/schema/primitive_inferrer.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'dry/core/cache'
+
+module Dry
+  module Schema
+    # PrimitiveInferrer is used internally by `Macros::Filled`
+    # for inferring a list of possible primitives that a given
+    # type can handle.
+    #
+    # @api private
+    class PrimitiveInferrer
```

Comment:
You know, we could even have it in dry-types, wdyt?

Reply:
Yes, I would like to see this at some point. There are probably more useful tools we could build into dry-types.